cmd/traefik/traefik.go

```diff
@@ -317,6 +276,64 @@ func run(globalConfiguration *configuration.GlobalConfiguration) {
 	}
 	svr.Wait()
 	log.Info("Shutting down")
+	logrus.Exit(0)
+}
+
+func configureLogging(globalConfiguration *configuration.GlobalConfiguration) {
+	// configure default log flags
+	fmtlog.SetFlags(fmtlog.Lshortfile | fmtlog.LstdFlags)
+
+	if globalConfiguration.Debug {
+		globalConfiguration.LogLevel = "DEBUG"
+	}
```

Comment:
this code should be in setEffectiveConfiguration ? WDYT

Reply:
I am not sure about that. The docs of `SetEffectiveConfiguration` states:

> SetEffectiveConfiguration adds missing configuration parameters derived from existing ones. It also takes care of maintaining backwards compatibility.

In my perception, this is actually not the case.  `LogLevel` is not a missing configuration parameter. It's meant to be this way.

Additionally `SetEffectiveConfiguration` is called _after_ `configureLogging`, and that is for the good reason that the logging configuration should happen as first thing in order to achieve what this PR tries to achieve. Log all messages in the desired format and write them to a file if it is configured.